### ISSUE TITLE
In some cases, lane arrows are ignored by vehicles.

### ISSUE THREAD

**Issue by c-lipka** [CitiesSkylinesMods/TMPE#471]
### Description
Certain (unknown to me) circumstances may cause one or more lane arrows to be ignored by all vehicles.

### Reproduction Steps
1. Load the savegame.
2. Locate the junction seen in the screenshot of that savegame (see description of that savegame).
3. Remove all lane connectors from the junction.
4. Note how most vehicles from the road previously managed by the lane connectors now try to turn left, disregarding the "right turn only" lane arrows still set on the road.

(Note that the lane connectors currently set on the intersection do not trigger the issue, but rather just mask it. The lane arrows were already broken before I added the lane connectors as a workaround.)

### Mod List
Please list the mods enabled when you noticed the bug.
- TM:PE 10.21.1 LABS [Krzychu1245]
- Destroy Chirper [react]
- Network Extensions 2 [Lazarus*Man]
- Total Autobudget [Zenya]
- Tree brush [Destroyer]
- Unlimited Oil And Ore

### Output Log

_Let me know if you can't reproduce the issue on your end using the savegame._

### Savegame (Optional)
https://steamcommunity.com/sharedfiles/filedetails/?id=1820184126

**Comment by c-lipka** [comment]
Well, I feel stupid now - looks like whatever the issue may be, saving and re-loading the city seems to be fixing it.
I'll leave the issue open, but don't be too surprised if the savegame doesn't exhibit the described behaviour...

**Comment by krzychu124** [comment]
Thanks for reporting that 😃 
Unfortunately I can confirm that something is wrong when outgoing segment has monorail tracks (at least in your savegame). I will try to check what is going on there and hopefully fix that bug.

**Comment by aubergine10** [comment]
Tagging: #293 , #295

**Comment by c-lipka** [comment]
Yes, at this point I actually feel stupid about having felt stupid :)
By now, I've added a couple more in-line monorail stations to my city with road junctions right at the stations, and get consistent failure of the lane arrows at such junctions (and saving and re-loading does not seem to do the trick on most of them).

Having done a series of small experiments, it seems to be that the issue is specific to junctions directly at the end of any "not-quite-a-road" structure; so far I've tested with (and in all cases could create the issue):

- in-line monorail stations (Monorail Station with Road, Monorail Bus Hub)
- toll booths (tested with Two-Way Toll Booth)

By comparison, a proper road with monorail tracks in the same place would not cause any problem.

Another thing I noticed is that the issue is sensitive to the order in which things are built: Plonking any of the above structures next to an existing T-junction did not affect existing lane arrow set-up, while adding another road to an existing junction involving any of the above structures would always cause that road's lane arrows to be completely ignored.

**Comment by krzychu124** [comment]
Thanks for extensive testing, I did similar experiments two days ago and came to the same conclusion. Definitely need to do more tests, gather some debug logs and analyze what is wrong there.

**Comment by aubergine10** [comment]
Could it be that altering a road next to a 'not quite a road' causes the 'not quite a road' to be updated but in such a way that it doesn't trigger TM:PE to detect the replaced 'not quite a road'?

**Comment by aubergine10** [comment]
Not sure if useful, but while looking at something else I noticed that `NetAI` (which can be accessed via `NetInfo.m_netAI`) contains this:

`GetNodeBuilding(ushort nodeID, ref NetNode data, out BuildingInfo building, out float heightOffset)`

If that gives details about buildings such as toll booths, metro stations, etc., it might be useful? If so, it would also be useful for anything else that needs to detect toll booths or similar "roads with buildings".
